File: 프로그래머스/0/120878. 유한소수 판별하기/유한소수 판별하기.java
class Solution {
    public int solution(int a, int b) {
        int answer = 0;
        
        int g = gcd(a, b);
        
        a /= g;
        b /= g;
        
        answer = fc(a, b);
        
        return answer;
    }
    
    public static int gcd(int a, int b){
        if(b == 0){
            return a;
        }
        return gcd(b, a % b);
    }
    
    public static int fc(int a, int b){
        while(b % 2 == 0 || b % 5 == 0){
            if(b % 2 == 0) b /= 2;
            if(b % 5 == 0) b /= 5;
        }
        return a % b == 0 ? 1 : 2;
    } 
}

File: 프로그래머스/0/120878. 유한소수 판별하기/유한소수 판별하기.py
def solution(a, b):
    answer = 0
    
    while True:
        if b % 5 == 0:
            b //= 5
        elif b % 2 == 0:
            b //= 2
        else:
            break
    
    answer = 1 if a % b == 0 else 2
    
    return answer